src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs

```diff
@@ -57,20 +60,21 @@ public class TournamentService : ITournamentService
         #region Query Objects
 
         private readonly IQuery<Tournament, UniqueTournamentCriteria> _uniqueTournamentQuery;
-        private readonly IQuery<List<Tournament>, GetAllCriteria> _getAllQuery;
+        private readonly IQuery<ICollection<Tournament>, GetAllCriteria> _getAllQuery;
         private readonly IQuery<Tournament, FindByIdCriteria> _getByIdQuery;
-        private readonly IQuery<List<Team>, GetAllCriteria> _getAllTeamsQuery;
-        private readonly IQuery<List<TeamTournamentDto>, FindByTournamentIdCriteria> _tournamentTeamsQuery;
-        private readonly IQuery<List<Division>, TournamentDivisionsCriteria> _getAllTournamentDivisionsQuery;
-        private readonly IQuery<List<Group>, DivisionGroupsCriteria> _getAllTournamentGroupsQuery;
+        private readonly IQuery<ICollection<Team>, GetAllCriteria> _getAllTeamsQuery;
+        private readonly IQuery<ICollection<TeamTournamentDto>, FindByTournamentIdCriteria> _tournamentTeamsQuery;
+        private readonly IQuery<ICollection<Division>, TournamentDivisionsCriteria> _getAllTournamentDivisionsQuery;
+        private readonly IQuery<ICollection<Group>, DivisionGroupsCriteria> _getAllTournamentGroupsQuery;
         private readonly IQuery<TournamentScheduleDto, TournamentScheduleInfoCriteria> _getTournamentDtoQuery;
         private readonly IQuery<Tournament, TournamentByGroupCriteria> _getTournamenrByGroupQuery;
-        private readonly IQuery<List<Tournament>, OldTournamentsCriteria> _getOldTournamentsQuery;
+        private readonly IQuery<ICollection<Tournament>, OldTournamentsCriteria> _getOldTournamentsQuery;
 
         #endregion
 
         #region Constructor
 
+#pragma warning disable S107 // Methods should not have too many parameters
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not suppress issues. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1309)